### ISSUE TITLE
valid-expect should treat .finally() as part of async assertion promise chains

### ISSUE THREAD

**Issue by nami8824** [vitest-dev/eslint-plugin-vitest#904]
## Summary

`valid-expect` currently follows async assertion promise chains when they continue with `.then()` or `.catch()`.

However, it does not treat `.finally()` the same way:

```ts
await expect(promise).resolves.toBe(1).finally(() => {})
```

This assertion is awaited at runtime, because the whole promise chain is awaited. But `valid-expect` only walks through `.then()` and `.catch()` chains, so it can report the inner async assertion as not awaited.

## Reproduction

Add the following cases to the valid section of `tests/valid-expect.test.ts.`

```ts
'test("valid-expect", async () => { await expect(Promise.resolve(1)).resolves.toBe(1).then(() => {}).finally(() => console.log("valid-case")); });',
```

Then run: `pnpm test tests/valid-expect.test.ts`.

Currently, this case fails.
